packages/react-core/src/filters/applyFiltersToTileContent.js

```diff
@@ -0,0 +1,82 @@
+import { buildBinaryFeatureFilter } from './Filter';
+
+export function applyFiltersToTileContent(tileContent, filters) {
+  if (!filters || !Object.keys(filters).length || !tileContent) {
+    return tileContent;
+  }
+
+  const filterFn = buildBinaryFeatureFilter({ filters });
+
+  const applyFiltersFn = (idx, data) =>
+    (!data.attributes || data.attributes.getFilterValue.value[idx]) &&
+    filterFn(idx, data);
+
+  const pointsGetFilterValue = tileContent.points.featureIds.value.map((_, idx) =>
+    applyFiltersFn(idx, tileContent.points)
+  );
+
+  const polygonsGetFilterValue = applyToPolygonsLinesData(
+    tileContent.polygons,
+    applyFiltersFn
+  );
+
+  const linesGetFilterValue = applyToPolygonsLinesData(tileContent.lines, applyFiltersFn);
+
+  return {
+    ...tileContent,
+    points: {
+      ...tileContent.pois,
+      attributes: {
+        getFilterValue: {
+          value: pointsGetFilterValue,
+          size: 1
+        }
+      }
+    },
+    polygons: {
+      ...tileContent.polygons,
+      attributes: {
+        getFilterValue: {
+          value: polygonsGetFilterValue,
+          size: 1
+        }
+      }
+    },
+    lines: {
+      ...tileContent.lines,
+      attributes: {
+        getFilterValue: {
+          value: linesGetFilterValue,
+          size: 1
+        }
+      }
+    }
+  };
+}
+
+// Aux
+function applyToPolygonsLinesData(data, applyFiltersFn) {
+  const featuresGetFilterValue = new Uint16Array(data.featureIds.value.length);
+
+  let idx = 0,
+    start = null;
+  for (const featureId of data.featureIds.value) {
```

Comment:
This solution works but it should be better if we do the loop over `primitivePolygonIndices` or `pathIndices` and you would not need to do this condition ```if (featureId === data.featureIds.value[idx + 1])``` or work with two index (idx and idx+1).

In any case we can change it next week, it's not a blocker

Reply:
Updated.